euphony/src/main/cpp/core/source/Base32.cpp

```diff
@@ -0,0 +1,78 @@
+#include "../Base32.h"
+#include <iomanip>
+#include <sstream>
+#include <cmath>
+
+using namespace Euphony;
+
+Base32::Base32(const HexVector &hexVectorSrc)
+: hexVector(hexVectorSrc) {}
+
+std::string Base32::getBaseString() {
+    std::stringstream ss;
+    int sum = 0;
+    int rest = hexVector.getSize() % 5;
+
+    int count = 0;
+    for(u_int8_t hex : hexVector) {
+        count++;
+        sum = (sum << 4) | hex;
+        if(count % 5 == rest) {
+            ss << bitsToBase32(sum);
+            sum = 0;
+        }
+    }
```

Comment:
How about like below?
```suggestion
    const int bitDivisionUnit = 5;
    int rest = hexVector.getSize() % bitDivisionUnit;
    if(!rest) rest += bitDivisionUnit;

    int count = 0;
    for(u_int8_t hex : hexVector) {
        sum = (sum << 4) | hex;
        if(++count == rest) {
            ss << bitsToBase32(sum);
            rest += bitDivisionUnit;
            sum = 0;
        }
    }
```

I haven't run the unit tests yet. It is just pseudo-code.
In case of above suggested one, it is only modular operation removal & common constant extraction :)
if there's some issue, let me know.

It seems that you implemented the code well by using bit-wise operations.
I will do an additional review after reading the code in more detail.

Reply:
It goes well! I think we can reduce unnecessary caculation and make code more intuititional, with your solution. I'll change my PR. Thanks for your suggestion.😀